python/lsst/pipe/analysis/coaddAnalysis.py

```diff
@@ -418,20 +430,25 @@ def runDataRef(self, patchRefList, subdir="", cosmos=None):
                                               hscRun=repoInfo.hscRun, wcs=repoInfo.wcs,
                                               aliasDictList=aliasDictList)
             plotKwargs.update(dict(zpLabel=self.zpLabel))
-            self.plotMatches(matches, repoInfo.filterName, filenamer, repoInfo.dataId, forcedStr=forcedStr,
-                             **plotKwargs)
+            plotList.append(self.plotMatches(matches, repoInfo.filterName, filenamer, repoInfo.dataId,
+                                             forcedStr=forcedStr, **plotKwargs))
 
         for cat in self.config.externalCatalogs:
             with andCatalog(cat):
                 matches = self.matchCatalog(forced, repoInfo.filterName, self.config.externalCatalogs[cat])
                 if matches is not None:
-                    self.plotMatches(matches, repoInfo.filterName, filenamer, repoInfo.dataId,
-                                     forcedStr=forcedStr, matchRadius=self.matchRadius,
-                                     matchRadiusUnitStr=self.matchRadiusUnitStr, **plotKwargs)
+                    plotList.append(self.plotMatches(matches, repoInfo.filterName, filenamer, repoInfo.dataId,
+                                                     forcedStr=forcedStr, matchRadius=self.matchRadius,
+                                                     matchRadiusUnitStr=self.matchRadiusUnitStr,
+                                                     **plotKwargs))
                 else:
                     self.log.warn("Could not create match catalog for {:}.  Is "
                                   "lsst.meas.extensions.astrometryNet setup?".format(cat))
 
+        self.allStats = {}
+        self.allStatsHigh = {}
+        self.allStats, self.allStatsHight = persistPlots(plotList, repoInfo.dataId, repoInfo.butler)
```

Comment:
What purpose does the  `self.allStats = {}` and `self.StatsHigh = {}` serve?

`persistPlots` should really return a Struct too.

Reply:
None, they were just a leftover from before persistPlots was its own function and never got removed.

I'm not sure I agree with you that if a function returns more than one value that it should be done with a struct. A two value tuple return is quite normal in my opinion. However if you are thinking to future proof this if there are ever more than two values, there may be a point there, but I would like to consider if there will be before packing everything inside another object.